CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc

```diff
@@ -162,6 +273,70 @@ static function DisableLockAndBreakthrough()
 	}
 }
 
+static function PatchWeaponTechs()
+{
+	AddPrototypeItem('MagnetizedWeapons', 'TLE_AssaultRifle_MG');
+	AddPrototypeItem('MagnetizedWeapons', 'TLE_Shotgun_MG');
+	AddPrototypeItem('MagnetizedWeapons', 'TLE_Pistol_MG');
+	AddPrototypeItem('GaussWeapons', 'TLE_SniperRifle_MG');
+	AddPrototypeItem('GaussWeapons', 'TLE_Cannon_MG');
+	AddPrototypeItem('AutopsyAdventStunLancer', 'TLE_Sword_MG');
+	
+	AddPrototypeItem('PlasmaRifle', 'TLE_AssaultRifle_BM');
+	AddPrototypeItem('AlloyCannon', 'TLE_Shotgun_BM');
+	AddPrototypeItem('PlasmaRifle', 'TLE_Pistol_BM');
+	AddPrototypeItem('PlasmaSniper', 'TLE_SniperRifle_BM');
+	AddPrototypeItem('HeavyPlasma', 'TLE_Cannon_BM');
+	AddPrototypeItem('AutopsyArchon', 'TLE_Sword_BM');
+
+	AddPrototypeItem('PlatedArmor', 'TLE_PlatedArmor');
+	AddPrototypeItem('PoweredArmor', 'TLE_PoweredArmor');
+}
+
+static function AddPrototypeItem(name TechName, name Prototype)
+{
+	local X2StrategyElementTemplateManager Manager;
+	local array<X2DataTemplate> DifficulityVariants;
+	local X2DataTemplate DataTemplate;
+	local X2TechTemplate TechTemplate;
+
+	Manager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	Manager.FindDataTemplateAllDifficulties(TechName, DifficulityVariants);
+	
+	foreach DifficulityVariants(DataTemplate)
+	{
+		TechTemplate = X2TechTemplate(DataTemplate);
+
+		if(TechTemplate != none)
+		{
+			if(TechTemplate.ResearchCompletedFn == none)
+			{
+				TechTemplate.ResearchCompletedFn = GiveAllItems;
+			}
+			TechTemplate.ItemRewards.AddItem(Prototype);
+		}
+	}
+}
+
+static function GiveAllItems(XComGameState NewGameState, XComGameState_Tech TechState)
```

Comment:
Why do we need both this and the event listener?

Reply:
Tried to call XComHQ's GiveItem function from inside the EventListenerReturn, but it didn't work with that function's GameState variable. So I went back here, where the OnResearchedFn delegate function has a NewGameState that can properly call GiveItem.

Reply:
It didn't wok as the passed gamestate is already submitted. You need to make a new one